pkg/log/klogcontrol/klogcontrol.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019-2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package klogcontrol
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+// Options captures runtime configuration for klog.
+type Options map[string]interface{}
+
+// Control implements runtime control for klog.
+type Control struct {
+	flags *flag.FlagSet
+}
+
+// Our singleton klog Control instance.
+var ctl *Control
+
+// Get returns our singleton klog Control instance.
+func Get() *Control {
+	return ctl
+}
+
+// CurrentOptions returns the current klog configuration as Options.
+func (c *Control) CurrentOptions() Options {
+	o := make(Options)
+	c.flags.VisitAll(func(f *flag.Flag) {
+		o[f.Name] = flag.Lookup(f.Name).Value.(flag.Getter).Get()
+	})
+	return o
+}
+
+// Configure reconfigures klog with the given Options.
+func (c *Control) Configure(options Options) error {
+	for name, value := range options {
+		if err := flag.Set(name, fmt.Sprintf("%v", value)); err != nil {
+			return klogError("failed to set klog flag %q to %v: %v", name, value, err)
+		}
+	}
+	return nil
+}
+
+// Set sets the value of the given klog flag.
+func (c *Control) Set(name, value string) error {
+	return flag.Set(name, value)
+}
+
+// Get returns the current value of the given klog flag.
+func (c *Control) Get(name string) (interface{}, error) {
+	if c.flags.Lookup(name) == nil {
+		return nil, klogError("unknown klog flag %q", name)
+	}
+	return flag.Lookup(name).Value.(flag.Getter).Get(), nil
+}
+
+// CloneFrom clones src to o.
+func (o *Options) CloneFrom(src Options) {
+	*o = make(Options)
+	for name, value := range src {
+		(*o)[name] = value
+	}
+}
+
+// String returns a string representation of the Options.
+func (o *Options) String() string {
+	if o == nil {
+		return "<nil>"
+	}
+	str := ""
+	sep := ""
+	for name, value := range *o {
+		str += sep + name + "=" + fmt.Sprintf("%v", value)
+		sep = "\n"
+	}
+	return str
+}
+
+// klogflag wraps a klog flag for configuration.
+type klogflag struct {
+	flag *flag.Flag
+}
+
+// Set implements flag.Value.Set() for wrapped klog flags.
+func (klogf *klogflag) Set(value string) error {
+	if klogf.flag.Name == "stderrthreshold" {
+		value = strings.ToUpper(value)
+	}
```

Comment:
What is this for?

Reply:
klog requires you to shout the logging threshold/level in ALL CAPS. I prefer to accept less aggressively formatted configuration, too. With this both (and admittedly all other combinations) are accepted.

Reply:
K, make sense. Maybe a short comment in the code would be in order because this just looks like a strange "anomaly". Or maybe it's just me. WDYT?

Reply:
Added a comment explaining the change to uppercase.